rdr_service/dao/workbench_dao.py

```diff
@@ -452,8 +452,24 @@ def get_workspaces_with_user_detail(self, **kwargs):
         now = clock.CLOCK.now()
         sequest_hours_ago = now - timedelta(hours=sequest_hour)
         with self.session() as session:
+            subquery = (
+                session.query(distinct(WorkbenchWorkspaceUser.workspaceId))
+                    .filter(WorkbenchWorkspaceUser.isCreator == 1)
+                    .subquery()
+            )
             count_query = (session.query(distinct(WorkbenchWorkspaceApproved.workspaceSourceId))
-                           .filter(WorkbenchWorkspaceApproved.excludeFromPublicDirectory == 0,
+                           .join(WorkbenchWorkspaceUser, WorkbenchResearcher, WorkbenchInstitutionalAffiliations)
+                           .filter(WorkbenchWorkspaceUser.workspaceId == WorkbenchWorkspaceApproved.id,
+                                   WorkbenchWorkspaceUser.researcherId == WorkbenchResearcher.id,
+                                   WorkbenchInstitutionalAffiliations.researcherId == WorkbenchResearcher.id,
+                                   WorkbenchWorkspaceApproved.excludeFromPublicDirectory == 0,
+                                   WorkbenchWorkspaceApproved.status == 1,
+                                   WorkbenchInstitutionalAffiliations.isVerified == 1,
+                                   or_(WorkbenchWorkspaceUser.isCreator == 1,
+                                       and_(
+                                           WorkbenchWorkspaceUser.role == 3,
```

Comment:
Can you use `int(WorkbenchWorkspaceUserRole.OWNER)` here instead of `3` to make it clearer what's being filtered?

Reply:
Good point, done!